src/ai/backend/manager/api/vfolder.py

```diff
@@ -1808,22 +1809,40 @@ async def delete(request: web.Request) -> web.Response:
 @auth_required
 @server_status_required(ALL_ALLOWED)
 @vfolder_permission_required(VFolderPermission.READ_ONLY)
-async def leave(request: web.Request, row: VFolderRow) -> web.Response:
+@check_api_params(
+    t.Dict(
+        {
+            tx.AliasedKey(["shared_user_uuid", "sharedUserUuid"], default=None): t.String | t.Null,
+        }
+    ),
+)
+async def leave(request: web.Request, params: Any, row: VFolderRow) -> web.Response:
     """
     Leave a shared vfolder.
 
     Cannot leave a group vfolder or a vfolder that the requesting user owns.
     """
     if row["ownership_type"] == VFolderOwnershipType.GROUP:
         raise InvalidAPIParameters("Cannot leave a group vfolder.")
-    if row["is_owner"]:
-        raise InvalidAPIParameters("Cannot leave a vfolder owned by the requesting user.")
 
     root_ctx: RootContext = request.app["_root.context"]
     access_key = request["keypair"]["access_key"]
-    user_uuid = request["user"]["uuid"]
+    user_role = request["user"]["role"]
+    rqst_user_uuid = request["user"]["uuid"]
+    shared_user_uuid = params["shared_user_uuid"]
     vfolder_id = row["id"]
-    perm = None
+    perm = row["permission"]
+
+    if shared_user_uuid:
+        # Allow only superadmin to leave the shared vfolder of others.
+        if (rqst_user_uuid != shared_user_uuid) and (user_role not in (UserRole.SUPERADMIN)):
```

Comment:
Uh... Is there a specific reason using `in` operator instead of `==`? Of course this will work, but still not a perfect way to do so. If you were intending to use `in` to check if `user_role` exists inside tuple containing single `UserRole.SUPERADMIN` element, then please add comma at the end of tuple element. The braces on this line will work as a brace to wrap code, not for defining tuple. 


```suggestion
        if (rqst_user_uuid != shared_user_uuid) and (user_role not in (UserRole.SUPERADMIN,)):
```

Reply:
It's superadmin only. So I changed to use `==` as you mentioned. Thanks!